eox_tenant/management/commands/create_or_update_tenant_config.py

```diff
@@ -0,0 +1,122 @@
+            type=str,
+            help='External key of the tenant config'
+        )
+        parser.add_argument('routes', nargs='+', help='Routes to link to this tenant config')
+
+        parser.add_argument(
+            '--config',
+            type=json.loads,
+            help="Enter JSON tenant configurations",
+            required=False
+        )
+        parser.add_argument(
+            '-f',
+            '--config-file',
+            type=load_json_from_file,
+            dest='config_file_data',
+            help="Enter the path to the JSON file containing the tenant configuration",
+            required=False
+        )
+
+    def merge_dict(self, base_dict: dict, override: dict):
+        """
+        Merge two nested dicts.
+        """
+        for key, value in override.items():
+            if isinstance(value, dict):
+                merged = base_dict.get(key, {}).copy()
+                merged.update(value)
+                base_dict[key] = merged
+                continue
+            base_dict[key] = value
+        return base_dict
+
+
+    def handle(self, *args, **options):
+        """
+        Create or update TenantConfig and link related routes.
+        """
+        external_key = options['external_key']
+        routes = options['routes']
+        configuration = options.get('config')
+        config_file_data = options.get('config_file_data')
+        tenant_configuration_values = configuration or config_file_data
+        # pylint: disable=no-member,protected-access
+        external_key_length = TenantConfig._meta.get_field("external_key").max_length
+        if external_key:
+            if len(str(external_key)) > external_key_length:
+                LOG.warning(
+                    "The external_key %s is too long, truncating to %s"
+                    " characters. Please update external_key in admin.",
+                    external_key,
+                    external_key_length
+                )
+            # trim name as the column has a limit of 63 characters
+            external_key = external_key[:external_key_length]
+        tenant, created = TenantConfig.objects.get_or_create(
+            external_key=external_key,
+        )
+        if created:
+            LOG.info("Tenant does not exist. Created new tenant: '%s'", tenant.external_key)
+        else:
+            LOG.info("Found existing tenant for: '%s'", tenant.external_key)
+
+        # split out lms, studio, theme, meta from configuration json
+        if tenant_configuration_values:
+            for field in TenantConfig._meta.get_fields():
+                if isinstance(field, JSONField):
+                    name = field.name
+                    value = tenant_configuration_values.get(name)
+                    if value:
+                        base_value = getattr(tenant, name, {})
+                        merged = self.merge_dict(base_value, value)
+                        setattr(tenant, name, merged)
```

Comment:
Currently we can only update config fields or add now config keys, but cannot remove an existing config key.

Perhaps we can introduce an optional boolean flag `--override` to completely remove the existing values instead of merging:
```suggestion
                        if override:
                            setattr(tenant, name, value)
                        else:
                            base_value = getattr(tenant, name, {})
                            merged = self.merge_dict(base_value, value)
                            setattr(tenant, name, merged)
```

Reply:
Updated.